src/instructlab/data/generate.py

```diff
@@ -210,8 +210,11 @@ def generate(
         from instructlab.model.backends import backends
         from instructlab.model.backends.llama_cpp import Server as llama_cpp_server
 
+        # TODO (cdoern): we really should not edit the cfg object
         ctx.obj.config.serve.llama_cpp.llm_family = model_family
-        backend_instance = backends.select_backend(ctx.obj.config.generate.teacher)
+        backend_instance = backends.select_backend(
+            cfg=ctx.obj.config.serve, model_path=model
+        )
```

Comment:
@cdoern should we add a check to make sure `model != default` before essentially overriding what is supplied in the ctx? To make sure we dont replace a custom mdoel with the default as @booxter pointed out

Reply:
nvm, sounds like this comment might be obsolete now